daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,333 @@
+package firewall
+
+import (
+	"errors"
+	"fmt"
+	"net/netip"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/NordSecurity/nordvpn-linux/daemon/device"
+)
+
+var ErrRuleAlreadyActive = errors.New("this rule is already active")
+
+const (
+	iptablesCommand  = "iptables"
+	ip6tablesCommand = "ip6tables"
+)
+
+// IptablesExecutor is an abstraction over iptables command. It should accept any input that is also viable for iptables
+// command.
+type IptablesExecutor interface {
```

Comment:
I know it would require some rework. But could this have a priority functionality. E. g. instead of always inserting to the top, we would need to calculate index based on priority and insert to the right place with `-I index`.  I see 2 options for this:
1. We can use `comment` module for storing priority in `iptables`. If we do this, we could have fixed priorities for each of the rule types and not care about correct function call order anymore. So it would look like this:
```go
InsertRule(command string, priority int) error
```
2. We can use `comment` module for specific key of a rule. E. g. `nordvpn_block_traffic` and then have a priority list somewhere in the executor. Then it would look like this:
```go
func NewIPTablesExecutor(order []string) *IPTablesExecutor { ... }
// ...
InsertRule(command string, key string) error
```
Then make sure that rule with a given key is always inserted in between the already existing rules with other keys in the order.

Another benefit of this would be that we don't need to provide a full rule to `DeleteRule` call and could have it as
```go
DeleteRule(priority int) error
// or
DeleteRule(key string) error
```

Reply:
Added a commit with priority tracking, to visualize how it would look like of such approach(didn't adjust uts yet though).

It increases complexity quite a bit. State management/book keeping in `FirewallManager` got more involved and `IptablesExecutor` got more complicated, so we'd probably have create a layer of abstraction over `Exec` calls so that it can be unit tested.

What is the benefit of using priority system over ensuring ordering in the implementation?

Rule deletion needs to stay the same, for some types of rules(meshnet peer permissions), we do not want to delete all the rules with the same key/priority. So we still need to provide specific rules.